### PR TITLE
Remove deprecated Vector::print overloads

### PR DESCRIPTION
**Pull request by masterleinad**
Deprecated in #991 and #1014.

### CODE REVIEW

File: tests/base/function_parser_05.cc [hunk 1]
```diff
@@ -85,7 +85,9 @@ test2()
 
   // Output the evaluated function
   deallog << "Function '" << expressions[0] << "," << expressions[1] << "'"
-          << " @ " << point << " is " << result << std::endl;
+          << " @ " << point << " is " << std::flush;
+  result.print(deallog.get_file_stream(), 4, false);
```
Comment (bangerth) [inline tests/base/function_parser_05.cc:89]: Is this a mistake? I thought you were *removing* the `print` function.

Comment (masterleinad) [reply tests/base/function_parser_05.cc:89]: We are removing the overload taking `LogStream` but we still allow `std::ostream`.

Comment (masterleinad) [reply tests/base/function_parser_05.cc:89]: Or are you saying that we shouldn't remove the `operator<<(LogStream,...)` function that is implemented in terms of `print` as well?

Comment (bangerth) [reply tests/base/function_parser_05.cc:89]: No, I was just confused why we are moving from `operator<<` to `results.print(...)`. I hadn't realized that you were also removing `operator<<(LogStream)`.

But if it were up to me, we'd completely remove `LogStream` to begin with...

**Pull request merged**